crates/swc_common/src/plugin.rs

```diff
@@ -8,38 +8,58 @@ use crate::{syntax_pos::Mark, SyntaxContext};
 use anyhow::Error;
 use std::any::type_name;
 
-/// Serialize ast to pass into plugins. This is being called from host side.
-/// TODO: https://github.com/swc-project/swc/issues/3167
-#[cfg(feature = "plugin-base")]
-pub fn serialize_for_plugin<T>(t: &T) -> Result<rkyv::AlignedVec, Error>
+/// A struct wraps internal representation of serialized Program to be passed
+/// across plugin's host to guest. Plugin author should not rely on specific
+/// details serialized byte format struct contains: it is strictly
+/// implementation detail which can change anytime.
+#[cfg(any(feature = "plugin-mode", feature = "plugin-rt"))]
+pub struct SerializedProgram(pub rkyv::AlignedVec);
```

Comment:
This is mostly to hide implementation detail from plugin author. We may change internal implementation any time, but plugin author should never care about specific data type we use (`AlignedVec`)

Reply:
I think `swc_common` is not a good place to store this, because breaking change of `swc_common` means breaking change of `swc_ecma_ast`. (Because of the way cargo works)